ImputationPipeline/notebooks/CreateManuallyQCdPRSResultsFile.ipynb

```diff
@@ -0,0 +1,859 @@
+    "            \n",
+    "    def manuallyUnFailConditionForSample(self, sample, condition):\n",
+    "        self.removeManualFailureFromResults(sample, condition)    \n",
+    "        self.manual_fails[sample].pop(condition)\n",
+    "        if len(self.manual_fails[sample]) == 0:\n",
+    "            self.manual_fails.pop(sample)\n",
+    "\n",
+    "    #when buttons clicked, fail or unfail sample for all conditions\n",
+    "    def resetSampleFailureWidgets(self):\n",
+    "        self.sample_failure_selection.value = ''\n",
+    "        self.sample_failure_reason_selection.value = ''\n",
+    "        self.sample_failure_reason_selection.disabled = True\n",
+    "        self.fail_sample_all_conditions_button.disabled = True\n",
+    "        self.unfail_sample_all_conditions_button.disabled = True\n",
+    "    \n",
+    "    def clickFailForAllConditionsButton(self, b):\n",
+    "        sample = self.sample_failure_selection.value\n",
+    "        reason = self.sample_failure_reason_selection.value\n",
+    "        self.manuallyFailSampleForAllConditions(sample, reason)\n",
+    "        self.resetSampleFailureWidgets()\n",
+    "    \n",
+    "    def clickUnfailForAllConditionsButton(self, b):\n",
+    "        sample = self.sample_failure_selection.value\n",
+    "        self.manuallyUnFailSampleForAllConditions(sample)\n",
+    "        self.resetSampleFailureWidgets()\n",
+    "\n",
+    "        \n",
+    "    def printFailedSamplesAllConditions(self):\n",
+    "        with self.out_failed_sample_all_conditions:\n",
+    "            clear_output()\n",
+    "            failed_samples_all_conditions_results = self.modified_results.loc[self.modified_results['poly_test_not_performed_reason']=='Failed PRS']\n",
+    "            if failed_samples_all_conditions_results.empty:\n",
+    "                 print('No Samples Will Be Failed For All Conditions')\n",
+    "            else:\n",
+    "                print('The following samples will be marked as Failed PRS and failed for all conditions')\n",
+    "                for row in failed_samples_all_conditions_results.iterrows():\n",
+    "                    print(row[0] + \", notes: \" + row[1]['notes'])\n",
+    "\n",
+    "            \n",
+    "    #third section, fail condition for all samples\n",
+    "    #third section displays when second section completed\n",
+    "    def clickFinishedSampleFailuresButton(self,button):\n",
+    "        self.sample_failure_hbox.close()\n",
+    "        self.finished_sample_failures_button.close()\n",
+    "        self.build_fail_condition_section()\n",
+    "        self.printFailedConditionsAllSamples()\n",
+    "\n",
+    "\n",
+    "\n",
+    "\n",
+    "    def build_fail_condition_section(self):\n",
+    "        self.out_failed_condition_all_samples = widgets.Output(layout={'border': '1px solid black'})\n",
+    "        self.condition_failure_selection = widgets.Dropdown(options = [None] + self.conditions,\n",
```

Comment:
One thing about the `None` option, if someone would select a condition and then go back to the empty option, they would still be able to enter a reason, and if they then click "Fail" the notebook crashes. A quick fix in `clickFailForAllSamplesButton` if the `None` option is selected will fix that, but that's also a very unlikely scenario in my opinion

Reply:
good find!